src/Tasks/Microsoft.NET.Build.Tasks/PublishAssembliesResolver.cs

```diff
@@ -31,6 +31,11 @@ public IEnumerable<ResolvedFile> Resolve(ProjectContext projectContext)
 
             foreach (LockFileTargetLibrary targetLibrary in projectContext.GetRuntimeLibraries(_privateAssetPackageIds))
             {
+                if (targetLibrary.Type == "project")
```

Comment:
The assemblies in the compile section are raised to MSBuild and eventually added to the `Reference` items in [ResolveLockFileReferences](https://github.com/dotnet/sdk/blob/f07cdf44feb2328ccb3c5e9f83f041c0a2542961/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets#L308). Could there be a problem (unrelated to this PR) with a project being added through ProjectReferences and its output simultaneously added to References?